semgrep-core/src/core/Rule.ml

```diff
@@ -196,12 +196,15 @@ type taint_spec = {
 }
 [@@deriving show]
 
+(* Method to combine extracted ranges within a file:
+    - either treat them as separate files; or
+    - concatentate them together
+*)
+type extract_reduction = Separate | Concat [@@deriving show]
```

Comment:
Extract_method? Extract_treatment_of_ranges? (I don't have good ideas I just don't love the current name, as I read more of this file though I think it's less important)

Reply:
I think the bigger question would be for the user facing yaml key which corresponds to this. I don't think the internal name of this type is significant.

Reply:
Do you want to hash this out now, or just merge this with a disposition to revisit naming there later?

Reply:
Merge it